src/com/company/Main.java

```diff
@@ -8,18 +8,40 @@
 
 
 public class Main {
-    public static void main(String[] args){
+    public static void main(String[] args) {
+
+        String txtFilePath = args[0];
+
+        File file = new File(txtFilePath);
+        if(file.length()==0)
+        {
+            System.out.println("Enter the valid path ");
+            System.exit(0);
+        }
+        isFileFormatSupported(file);
+        isFileExists(file);
+
+        if((isFileExists(file) && isFileFormatSupported(file))== false)
+        {
+            System.out.println("file is not valid");
+            System.exit(0);
+        }
+
 
-        String txtfile = args[0];
-        File file = new File(txtfile);
-        checkForSupport(file);
-        checkForExistance(file);
-        String searchword = args[1];
         System.out.println("processing................");
-        String data = null;
-        data = readFileAsString(txtfile);
-        avoidSpecialChar(data);
-        searchTheWord(data,searchword);
+
+        String data = readFileAsString(txtFilePath);
+
+
+        String searchword = null;
+        try {
+            searchword = args[1];
```

Comment:
Check this at the starting of the program. File path and search word are mandatory for you program to execute. So if they are not present, then you don't need to continue further.

Reply:
yeah, I have made that with the if statement.